packages/amplify-graphql-relational-transformer/src/resolver/ddb-references-generator.ts

```diff
@@ -0,0 +1,360 @@
+const PARTITION_KEY_VALUE = 'partitionKeyValue';
+
+export class DDBRelationalReferencesResolverGenerator extends RelationalResolverGenerator {
+  makeExpression = (keySchema: any[], connectionAttributes: string[]): ObjectNode => {
+    if (keySchema[1] && connectionAttributes[1]) {
+      let condensedSortKeyValue;
+
+      if (connectionAttributes.length > 2) {
+        const rangeKeyFields = connectionAttributes.slice(1);
+
+        condensedSortKeyValue = rangeKeyFields
+          .map((keyField, idx) => `\${${SORT_KEY_VALUE}${idx}}`)
+          .join(ModelResourceIDs.ModelCompositeKeySeparator());
+      }
+
+      return obj({
+        expression: str('#partitionKey = :partitionKey AND #sortKey = :sortKey'),
+        expressionNames: obj({
+          '#partitionKey': str(keySchema[0].attributeName),
+          '#sortKey': str(keySchema[1].attributeName),
+        }),
+        expressionValues: obj({
+          ':partitionKey': ref(`util.dynamodb.toDynamoDB($${PARTITION_KEY_VALUE})`),
+          ':sortKey': ref(`util.dynamodb.toDynamoDB(${condensedSortKeyValue ? `"${condensedSortKeyValue}"` : `$${SORT_KEY_VALUE}0`})`),
+        }),
+      });
+    }
+
+    return obj({
+      expression: str('#partitionKey = :partitionKey'),
+      expressionNames: obj({
+        '#partitionKey': str(keySchema[0].attributeName),
+      }),
+      expressionValues: obj({
+        ':partitionKey': ref(`util.dynamodb.toDynamoDB($${PARTITION_KEY_VALUE})`),
+      }),
+    });
+  };
+
+  /**
+   * Create a resolver that queries an item in DynamoDB.
+   * @param config The connection directive configuration.
+   * @param ctx The transformer context provider.
+   */
+  makeHasManyGetItemsConnectionWithKeyResolver = (config: HasManyDirectiveConfiguration, ctx: TransformerContextProvider): void => {
+    const { connectionFields, field, indexName, limit, object, references, relatedType } = config;
+    const connectionAttributes: string[] = references.length > 0 ? references : connectionFields;
+    if (connectionAttributes.length === 0) {
+      throw new Error('Either connection fields or references should be populated.');
+    }
+
+    const primaryKeyFields: string[] = getPrimaryKeyFields(object);
+    const table = getTable(ctx, relatedType);
+    const dataSourceName = getModelDataSourceNameForTypeName(ctx, relatedType.name.value);
+    const dataSource = ctx.api.host.getDataSource(dataSourceName);
+    const keySchema = getKeySchema(table, indexName);
+
+    const setup: Expression[] = [
+      set(ref('limit'), ref(`util.defaultIfNull($context.args.limit, ${limit})`)),
+      ...primaryKeyFields
+        .slice(1)
+        .map((ca, idx) =>
+          set(
+            ref(`${SORT_KEY_VALUE}${idx}`),
+            methodCall(ref('util.defaultIfNull'), ref(`ctx.stash.connectionAttibutes.get("${ca}")`), ref(`ctx.source.${ca}`)),
+          ),
+        ),
+      set(ref('query'), this.makeExpression(keySchema, connectionAttributes)),
```

Comment:
Also this `connectionAttributes` passed into the `makeExpression`.  In the ref relation, this refers to reference fields on related model, which is different from the old way. I am trying to understand which is the right param to be passed, which is based on the following comment

Reply:
> In the ref relation, this refers to reference fields on related model, which is different from the old way.

For `@hasMany`, references (new way) behaves the same way as fields (old way) -- the `hasMany` field resolver queries the related model's table for the primary model's keys.